src/backend/revoker/indexer/queryFromIndexer.test.ts

```diff
@@ -0,0 +1,297 @@
+    offset: ${offset}
+    ${filter ? `filter: { ${filter} }` : ''}
+  ) {
+    totalCount
+    nodes {
+      ...wholeBlock
+    }
+  }
+}
+  ${wholeBlock}
+`;
+}
+/** Expected structure of responses for queries defined above. */
+interface IndexedBlock {
+  id: string; // Block Ordinal Number, without punctuation
+  hash: HexString;
+  timeStamp: string; // ISO8601 Date String, like 2022-02-09T13:09:18.217
+}
+
+function mockBlocks(numberOfBlocks: number) {
+  const mockedBlocks: IndexedBlock[] = [];
+
+  for (let index = numberOfBlocks; index > 0; index--) {
+    mockedBlocks.push({
+      id: index.toString(),
+      hash: `0x${index.toString(16)}`,
+      timeStamp: new Date(index, 3, 21).toISOString(),
+    });
+  }
+
+  return mockedBlocks;
+}
+
+jest.mock('../../utilities/configuration', () => ({
+  configuration: {
+    indexer: { graphqlEndpoint: 'https://dev-indexer.kilt.io/' },
+  },
+}));
+
+let postResponse: FetchedData;
+jest.mock('got', () => ({
+  post: jest.fn().mockReturnValue({
+    json: () => postResponse,
+  }),
+}));
+
+beforeEach(() => {
+  jest.mocked(got.post).mockClear();
+  configuration.indexer.graphqlEndpoint = 'https://dev-indexer.kilt.io/';
+});
+
+describe('The fundamental function to query from the Indexer', () => {
+  describe('queryFromIndexer()', () => {
+    describe('on positive cases', () => {
+      it('should query the KILT Indexer API', async () => {
+        postResponse = {
+          data: {
+            blocks: {
+              totalCount: 7_000_000,
+              nodes: [
+                {
+                  id: '7000000',
+                  timeStamp: '2024-09-16T11:48:12',
+                  hash: '0xCAFE',
+                },
+                {
+                  id: '6999999',
+                  timeStamp: '2024-09-16T10:48:12',
+                  hash: '0xCAB',
+                },
+                {
+                  id: '6999998',
+                  timeStamp: '2024-09-16T9:48:12',
+                  hash: '0xC10BE11A',
+                },
+              ],
+            },
+          },
+        };
+
+        const queryBlocks = buildBlockQueries()(0);
+
+        await queryFromIndexer(queryBlocks);
+
+        expect(got.post).toHaveBeenCalledWith(
+          configuration.indexer.graphqlEndpoint,
+          {
+            json: {
+              query: queryBlocks,
+            },
+          },
+        );
+      });
+      it('should return empty array when matches count is 0', async () => {
+        postResponse = {
+          data: {
+            blocks: {
+              totalCount: 0,
+              nodes: [],
+            },
+          },
+        };
+
+        const queryBlocks = buildBlockQueries(
+          'timeStamp: { greaterThan: "3000-6-9" }',
+        )(0);
+
+        const unmatched = await queryFromIndexer(queryBlocks);
+
+        expect(unmatched.totalCount).toBe(0);
+        expect(unmatched.matches).toBeDefined();
+        expect(unmatched.matches.length).toBe(0);
```

Comment:
Would this be more readable if we used `toMatchObject`?

Reply:
Sure.

Reply:
did it

Reply:
What do you think of this?

```ts
        expect(unmatched).toMatchObject({ totalCount: 0, matches: [] });

Reply:
I like it separated better.

Reply:
Catching bugs in a granular way seems more sensible to me.